### PR TITLE
MODFEE-220: Make paymentStatus property in accountdata.json required and also make it an enum

### PR DESCRIPTION
**Pull request by yusuf-murodov-gamerpay**
## Purpose
[MODFEE-220](https://issues.folio.org/browse/MODFEE-220)
_Make paymentStatus property in accountdata.json required and also make it an enum_

### CODE REVIEW

File: ramls/accountdata.json [hunk 1]
```diff
@@ -42,7 +42,8 @@
       "type": "object",
       "properties": {
         "name": {
-          "description": "Name of the status (values used are Outstanding, Paid partially, Paid fully, Waived partially, Waived fully, Transferred partially, Transferred fully, Refunded partially, Refunded fully, Cancelled as error)",
+          "description": "Name of the status",
+          "enum": ["Outstanding", "Paid partially", "Paid fully", "Waived partially", "Waived fully", "Transferred partially", "Transferred fully", "Refunded partially", "Refunded fully", "Credited fully", "Credited partially", "Cancelled item returned", "Cancelled as error"],
```
Comment (OleksandrVidinieiev) [inline ramls/accountdata.json:46]: Check possible values

File: src/main/java/org/folio/rest/service/action/CancelActionService.java [hunk 2]
```diff
@@ -52,7 +52,7 @@ protected Feefineaction createFeeFineActionAndUpdateAccount(Account account, Mon
       .withId(UUID.randomUUID().toString())
       .withDateAction(new Date());
 
-    account.getPaymentStatus().setName(reasonForAction);
+    account.getPaymentStatus().setName(PaymentStatus.Name.fromValue(reasonForAction));
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/rest/service/action/CancelActionService.java:55]: static import

File: src/main/java/org/folio/rest/service/action/RefundActionService.java [hunk 3]
```diff
@@ -161,7 +162,7 @@ private Feefineaction buildFeeFineAction(Account account, Action action, Monetar
 
   private static void updateAccountInMemory(Account account, Feefineaction feeFineAction) {
     account.setRemaining(feeFineAction.getBalance());
-    account.getPaymentStatus().setName(feeFineAction.getTypeAction());
+    account.getPaymentStatus().setName(PaymentStatus.Name.fromValue(feeFineAction.getTypeAction()));
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/rest/service/action/RefundActionService.java:165]: static import?

File: src/test/java/org/folio/rest/domain/AccountPaymentStatusTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+import org.junit.runner.RunWith;
+
+import junitparams.JUnitParamsRunner;
+import junitparams.Parameters;
+
+@RunWith(JUnitParamsRunner.class)
+public class AccountPaymentStatusTest {
+
+  @Test
+  @Parameters({"Outstanding", "Paid partially", "Paid fully", "Waived partially", "Waived fully",
```
Comment (OleksandrVidinieiev) [inline src/test/java/org/folio/rest/domain/AccountPaymentStatusTest.java:22]: Double check this list. And fix indentation.

File: src/test/java/org/folio/rest/domain/AccountPaymentStatusTest.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+               "Transferred partially", "Transferred fully", "Refunded partially", "Refunded fully",
+               "Credited fully", "Credited partially", "Cancelled item returned", "Cancelled as error"
+  })
+  public void paymentStatusIsValid(String status) {
```
Comment (OleksandrVidinieiev) [inline src/test/java/org/folio/rest/domain/AccountPaymentStatusTest.java:26]: add blank line

File: src/test/java/org/folio/rest/domain/AccountPaymentStatusTest.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    assertNotNull(account);
+    assertEquals(status, account.getPaymentStatus().getName().value());
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void paymentStatusIsNotValid() {
+    buildAccount("Not paid");
+  }
+
+  private Account buildAccount(String status) {
```
[the remark this answers is not in the document]
Comment (yusuf-murodov-epam) [reply src/test/java/org/folio/rest/domain/AccountPaymentStatusTest.java]: replaced to EntityBuilder

File: src/test/java/org/folio/rest/domain/AccountPaymentStatusTest.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+  })
+  public void paymentStatusIsValid(String status) {
+    Account account = buildAccount(status);
+    assertNotNull(account);
+    assertEquals(status, account.getPaymentStatus().getName().value());
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void paymentStatusIsNotValid() {
+    buildAccount("Not paid");
```
[the remark this answers is not in the document]
Comment (yusuf-murodov-epam) [reply src/test/java/org/folio/rest/domain/AccountPaymentStatusTest.java]: refactored

File: ramls/accountdata.json (hunk 1 above)
Comment (yusuf-murodov-epam) [reply ramls/accountdata.json]: checked and added

File: src/test/java/org/folio/rest/domain/AccountPaymentStatusTest.java [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+import org.folio.rest.jaxrs.model.Account;
+import org.folio.rest.jaxrs.model.PaymentStatus;
+import org.folio.test.support.EntityBuilder;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+class AccountPaymentStatusTest {
+
```
Comment (OleksandrVidinieiev) [inline src/test/java/org/folio/rest/domain/AccountPaymentStatusTest.java:16]: If `fromValue()` is the method that throws `IllegalArgumentException` when invalid `PaymentStatus` is supplied, then calling `fromValue` **only** should be sufficient to test both cases - no need to build the Account.

**Comment by OleksandrVidinieiev** [comment]
@yusuf-murodov-epam Please add ticket name to your future commit messages

File: src/test/java/org/folio/rest/domain/AccountPaymentStatusTest.java (hunk 8 above)
Comment (yusuf-murodov-epam) [reply src/test/java/org/folio/rest/domain/AccountPaymentStatusTest.java:16]: fixed

**Comment by OleksandrVidinieiev** [comment]
> > Please add ticket name to your future commit messages
> 
> do you mean on the first commit of pr or all the commits when I make changes?

All commits should have the ticket name. Otherwise it's hard to track changes during module release.

**Pull request merged**